### PR TITLE
Adds validation for model name

### PR DESCRIPTION
**Pull request by jasoncharnes**
#348

### CODE REVIEW

File: test/commands/generate/model_test.rb [hunk 1]
```diff
@@ -9,6 +9,10 @@
       -> { Lotus::Commands::Generate::Model.new({}, '') }.must_raise ArgumentError
       -> { Lotus::Commands::Generate::Model.new({}, '   ') }.must_raise ArgumentError
     end
+
+    it 'validates model name' do
+      -> { Lotus::Commands::Generate::Model.new({}, '123') }.must_raise ArgumentError
```
Comment (joneslee85) [inline test/commands/generate/model_test.rb]: what do you think about asserting the exception message here too?

```
exception = -> { Lotus::Commands::Generate::Model.new({}, '123') }.must_raise ArgumentError
exception.message.must_equal 'The error message'
```

File: lib/lotus/commands/generate/model.rb [hunk 2]
```diff
@@ -35,6 +36,14 @@ def assert_model_name!
           end
         end
 
+        # @since x.x.x
+        # @api private
+        def assert_model_has_valid_name!
+          unless model_name.match(/^[a-z]/i)
+            raise ArgumentError.new("Invalid model name")
```
Comment (joneslee85) [inline lib/lotus/commands/generate/model.rb]: What do you think about providing extra details on what a valid naming is?

Comment (joneslee85) [inline lib/lotus/commands/generate/model.rb]: also can you please add doc comment for `assert_model_name!`?

**Comment by jasoncharnes** [comment]
@joneslee85 appreciate the feedback, and I agree. I pushed an updated commit with the changes.

**Comment by jodosha** [comment]
@jasoncharnes Thanks for this fix.
Do you and @joneslee85 think it's a good idea to merge `assert_model_name!` with `assert_model_has_valid_name!`? as they both verify if the given name meets a set of rules: non-empty, no numbers.

**Comment by jasoncharnes** [comment]
@jodosha I was actually thinking about that while I adding `assert_model_has_valid_name!`. I'd be glad to update it if @joneslee85 agrees.

**Comment by joneslee85** [comment]
@jasoncharnes please do

**Comment by jasoncharnes** [comment]
@jodosha @joneslee85 let me know if this is/isn't along the lines of what your were thinking. Really digging Lotus! Thx for letting me get involved.

**Pull request merged**